File: python/word-count/wordcount.py
# A program that counts the occurrences of each word in a phrase

import re

def word_count(s):
    output = {}
    for word in re.findall(ur'[^\W_]+',s.lower().decode('utf-8'),flags=re.U):
        if word:
            output[word] = output.get(word, 0) + 1
    return output

File: java/word-count/src/main/java/WordCount.java
import java.util.*;

public class WordCount {
    public Map<String, Integer> phrase (String str) {
        String[] words = str.split("\\W+");
        Map<String, Integer> counts = new HashMap<>();

        for (String word : words)
            counts.merge(word.toLowerCase(), 1, (a, b) -> a + b);
        return counts;
    }
}
